UGENE-7652 Crash when exporting consensus from AE + adding it to MSA

rasputinkirill opened the pull request.

rasputinkirill committed:
UGENE-7652 Crash when exporting consensus from AE + adding it to MSA

rasputinkirill committed:
Merge remote-tracking branch 'remotes/origin/master' into UGENE-7652

# Conflicts:
#	src/plugins/GUITestBase/src/GUITestBasePlugin.cpp
#	src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
#	src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.h

yalgaer commented on src/corelibs/U2Gui/src/OpenViewTask.cpp:
```diff
@@ -84,13 +84,16 @@ static Task* createOpenViewTask(const MultiGSelection& ms) {
+        coreLog.info(QObject::tr("Unable to open view because of active modal widget."));
+        return t;
+    }
     if (ls.size() == 1) {
```
nit: all lines below can be a 1 liner: `return ls.size() == 1 ?  ls[0]->createViewTask()  : nullptr;`

yalgaer commented on src/corelibs/U2Gui/src/OpenViewTask.cpp:
```diff
@@ -84,13 +84,16 @@ static Task* createOpenViewTask(const MultiGSelection& ms) {
-
+    Task* t = nullptr;
+    if (QApplication::activeModalWidget() != nullptr) {
+        coreLog.info(QObject::tr("Unable to open view because of active modal widget."));
```
This is a big limitation and we should allow creation of views in the background.
May be it worth to create a separate bug for that.

yalgaer commented:
The fix looks reasonable. Please run GUI tests and I will approve.

rasputinkirill committed:
review issues

yalgaer commented:
Too many tests failed on Linux, please fix

rasputinkirill committed:
GUI tests update
Merge remote-tracking branch 'remotes/origin/master' into UGENE-7652

# Conflicts:
#	src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

rasputinkirill committed:
GUI test update
